File: client/src/modules/readjson/BuffTarget.js
/**
 * Created by Team 2 - LongLH - GDF 17 on 12/12/2022.
 */


var BuffTarget = Buff.extend({

    moveDistance: Map.prototype,

    getDelay: function (evolution) {
        return this.effects[evolution][0].delay;
    },

    getHpPerDelay: function (evolution, level) {
        let hpPerDelay = this.effects[evolution][0].hpPerDelay;
        if (this.effectsUseCardLevel) {
            hpPerDelay *= 1 + 0.1 * (level - 1);
        }
        return hpPerDelay;
    },

});

File: server/src/readjson/BuffTarget.java
package readjson;

import java.util.ArrayList;
import java.util.HashMap;

public class BuffTarget extends Buff {
    private HashMap<Integer, Double> moveDistance;

    public double getDelay(int evolution) {
        return this.effects.get(evolution).get(0).delay;
    }

    public double getHpPerDelay(int evolution, int level) {
        double hpPerDelay = this.effects.get(evolution).get(0).hpPerDelay;
        if (this.effectsUseCardLevel) {
            hpPerDelay *= 1 + 0.1 * (level - 1);
        }
        return hpPerDelay;
    }

}

